src/main.qml

```diff
@@ -517,4 +517,130 @@ ApplicationWindow
+            result.scheme = schemeStr.substring(0,j)
+
+            j += sepStr.length
+            schemeStr = schemeStr.substring(j) // from j to end
+
+            j = schemeStr.indexOf("?")
+            result.action = schemeStr.substring(0, j)
+
+            schemeStr = schemeStr.substring(j+1)
+
+            j = schemeStr.indexOf("#")
+            if (j >= 0) {
+                result.comment = schemeStr.substring(j+1)
+               schemeStr = schemeStr.substring(0, j)
+            }
+
+            // split our query string into its component parts
+            var arr = schemeStr.split('&');
+
+            for (var i = 0; i < arr.length; i++) {
+                // separate the keys and the values
+                var a = arr[i].split('=');
+
+                // set parameter name and value (use 'true' if empty)
+                var paramName = a[0];
+                var paramValue = typeof (a[1]) === 'undefined' ? true : a[1];
+                if (decode && typeof (paramValue) === typeof ("")) // if value is string
+                    paramValue = decodeURIComponent(paramValue)
+
+                // if the paramName ends with square brackets, e.g. colors[] or colors[2]
+                if (paramName.match(/\[(\d+)?\]$/)) {
+
+                    // create key if it doesn't exist
+                    var key = paramName.replace(/\[(\d+)?\]/, '');
+                    if (!queryParams[key]) queryParams[key] = [];
+
+                    // if it's an indexed array e.g. colors[2]
+                    if (paramName.match(/\[\d+\]$/)) {
+                        // get the index value and add the entry at the appropriate position
+                        var index = /\[(\d+)\]/.exec(paramName)[1];
+                        queryParams[key][index] = paramValue;
+                    } else {
+                        // otherwise add the value to the end of the array
+                        queryParams[key].push(paramValue);
+                    }
+                } else {
+                    // we're dealing with a string
+                    if (!queryParams[paramName]) {
+                        // if it doesn't exist, create property
+                        queryParams[paramName] = paramValue;
+                    } else if (queryParams[paramName] && typeof queryParams[paramName] === 'string'){
+                        // if property does exist and it's a string, convert it to an array
+                        queryParams[paramName] = [queryParams[paramName]];
+                        queryParams[paramName].push(paramValue);
+                    } else {
+                        // otherwise add the property
+                        queryParams[paramName].push(paramValue);
+                    }
+                }
+            }
+        }
+        result.parameters = queryParams
+        return result;
+    }
+
+    function startSearch(address) {
+        if (address > "") {
+            header.searching = true;
```

Comment:
I would have done this a bit differently, if you don't mind:
in `PlaceHeader.qml`, add a function:
```
function search(text) {
                searching = true
                resultList.model = undefined
                searchRequest(text)
}
```
and modify the `EnterKey.onClicked` handler in the PageHeader to call this function.

Then in `Header.qml`, add a forwarding function:
```
function search(text) {
  placeHeader.search(text)
}
```
and call it in the new `startSearch()` method of main you're introducing.

Reply:
I modified this a bit to write the address to the search field also.
```
    function search(text, changeText) {
        if (changeText)
            search.text = text // address from command line
```